options-user.php

```diff
@@ -1,39 +1,41 @@
 <?php
-// functions for managing Shibboleth user options through the WordPress administration panel
-
-add_action('profile_personal_options', 'shibboleth_profile_personal_options');
-add_action('personal_options_update', 'shibboleth_personal_options_update');
-add_action('show_user_profile', 'shibboleth_show_user_profile');
-add_action('admin_footer-user-edit.php', 'shibboleth_admin_footer_edit_user');
-
 /**
- * For WordPress accounts that were created by Shibboleth, limit what profile
- * attributes they can modify.
+ * For WordPress accounts that were created by Shibboleth, limit what administrators and users 
+ * can edit via user-edit.php and profile.php.
  *
- * @since 1.3
+ * @since 2.3
  */
-function shibboleth_profile_personal_options() {
-	$user = wp_get_current_user();
+function shibboleth_edit_user_options() {
+	if ( IS_PROFILE_PAGE ) {
+		$user_id = wp_get_current_user()->ID;
+	} else {
+		global $user_id;
+	}
 
-	if (get_user_meta( $user->ID, 'shibboleth_account') ) {
+	if ( get_user_meta( $user_id, 'shibboleth_account' ) ) {
+		add_filter( 'show_password_fields', '__return_false' );
 
-		add_filter( 'show_password_fields', create_function( '$v', 'return false;' ) );
+		add_action( 'admin_footer-user-edit.php', 'shibboleth_disable_managed_fields' );
 
-		add_action( 'admin_footer-profile.php', 'shibboleth_admin_footer_profile' );
+		add_action( 'admin_footer-profile.php', 'shibboleth_disable_managed_fields' );
 	}
 }
+add_action( 'personal_options', 'shibboleth_edit_user_options' );
 
 /**
  * For WordPress accounts that were created by Shibboleth, disable certain fields
- * that they are allowed to modify.
+ * that users/administrators aren't allowed to modify.
  *
- * @since 1.3
+ * @since 1.3 (renamed in 2.3 from `shibboleth_admin_footer_edit_user`)
```

Comment:
It seems like the function is more similar to the footer_profile, but merges most of the admin-only functionality that is relevant, so it's a combination.
```suggestion
 * @since 1.3 (renamed in 2.3 from `shibboleth_admin_footer_profile`)
```